lib/value/src/kind/collection.rs

```diff
@@ -0,0 +1,403 @@
+                    want: false,
+                },
+            ),
+            (
+                "unknown match",
+                TestCase {
+                    this: Collection::unknown(Kind::regex().or_null()),
+                    other: Collection::unknown(Kind::regex()),
+                    want: true,
+                },
+            ),
+            (
+                "unknown mis-match",
+                TestCase {
+                    this: Collection::unknown(Kind::regex().or_null()),
+                    other: Collection::unknown(Kind::bytes()),
+                    want: false,
+                },
+            ),
+            (
+                "other-known match",
+                TestCase {
+                    this: Collection::from_parts(
+                        BTreeMap::from([("bar", Kind::bytes())]),
+                        Kind::regex().or_null(),
+                    ),
+                    other: Collection::from_parts(
+                        BTreeMap::from([("foo", Kind::regex()), ("bar", Kind::bytes())]),
+                        Kind::regex(),
+                    ),
+                    want: true,
+                },
+            ),
+            (
+                "other-known mis-match",
+                TestCase {
+                    this: Collection::from_parts(
+                        BTreeMap::from([("foo", Kind::integer()), ("bar", Kind::bytes())]),
+                        Kind::regex().or_null(),
+                    ),
+                    other: Collection::from_parts(
+                        BTreeMap::from([("foo", Kind::regex()), ("bar", Kind::bytes())]),
+                        Kind::regex(),
+                    ),
+                    want: false,
+                },
+            ),
+            (
+                "self-known match",
+                TestCase {
+                    this: Collection::from_parts(
+                        BTreeMap::from([
+                            ("foo", Kind::bytes().or_integer()),
+                            ("bar", Kind::bytes().or_integer()),
+                        ]),
+                        Kind::bytes().or_integer(),
+                    ),
+                    other: Collection::unknown(Kind::bytes().or_integer()),
+                    want: true,
+                },
+            ),
+            (
+                "self-known mis-match",
+                TestCase {
+                    this: Collection::from_parts(
+                        BTreeMap::from([("foo", Kind::integer()), ("bar", Kind::bytes())]),
+                        Kind::bytes().or_integer(),
+                    ),
+                    other: Collection::unknown(Kind::bytes().or_integer()),
+                    want: false,
+                },
+            ),
+        ]) {
+            assert_eq!(this.is_superset(&other), want, "{}", title);
+        }
+    }
+
+    #[test]
+    #[allow(clippy::too_many_lines)]
+    fn test_merge() {
```

Comment:
Good catch. I added these tests before I added that strategy (since it was needed to pass some VRL tests). I updated the test case by copy/pasting the same strategy into all cases, but forgot to add new cases testing the different strategies. Will do so.